js/src/common/Application.js

```diff
@@ -331,19 +331,31 @@ export default class Application {
       }
 
       const isDebug = app.forum.attribute('debug');
+      // contains a formatted errors if possible, response must be an JSON API array of errors
+      // the details property is decoded to transform escaped characters such as '\n'
+      const formattedError = error.response && Array.isArray(error.response.errors) && error.response.errors.map(e => decodeURI(e.detail));
 
       error.alert = new Alert({
         type: 'error',
         children,
         controls: isDebug && [
-          <Button className="Button Button--link" onclick={this.showDebug.bind(this, error)}>Debug</Button>
+          <Button className="Button Button--link" onclick={this.showDebug.bind(this, error, formattedError)}>Debug</Button>
         ]
       });
 
       try {
         options.errorHandler(error);
       } catch (error) {
```

Comment:
Woah, can we rename this `error` variable in the `catch` block? I find it super confusing to follow along which object we're dealing with here. :grin: 

---

:clock1: Three minutes later...

---

Okay, I see now, the default `errorHandler` implementation just re-throws the same object, which means we have an AJAX error, which makes sense.

Could we change this to check for the existence of `options.errorHandler()` instead? The `try`/`catch` only confuses the reader IMO.

Reply:
Done. Had to remove the default `errorHandler` implementationa as it's not needed. 

I noticed that when the error handler is set - e.g. modals using `this.onerror` - nothing logs to the console. I assume we want to log the error to the console regardless if the error handler is set?

Reply:
Hmm, no, I think we should reserve the logging to unexpected errors - exactly the way you did it.

Really, it depends on whether we're dealing with a known or unknown error, but we have no mechanism for that yet (in custom error handlers).